### PR TITLE
MM-15645: go modules

### PR DESCRIPTION
**Pull request by lieut-data**
Switch over to go modules to work around dep failures at the moment and fix up incompatibilities with mattermost-server along the way. 

While we'll use go modules for dependency management, we're taking the same tack as mattermost-server and vendoring all dependencies to ease the burden on users getting started.

Fixes: https://mattermost.atlassian.net/browse/MM-15645

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	github.com/imdario/mergo v0.3.7 // indirect
+	github.com/inconshreveable/mousetrap v1.0.0 // indirect
+	github.com/jmoiron/sqlx v1.2.0
+	github.com/json-iterator/go v1.1.6 // indirect
+	github.com/kisielk/errcheck v1.2.0 // indirect
+	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
+	github.com/lib/pq v1.1.1
+	github.com/magiconair/properties v1.8.1 // indirect
+	github.com/mailru/easyjson v0.0.0-20190403194419-1ea4449da983 // indirect
+	github.com/mattermost/mattermost-server v5.11.0+incompatible
```
Comment (lieut-data) [inline go.mod:47]: Using 5.11 for now.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
+	gopkg.in/yaml.v2 v2.2.2
+	honnef.co/go/tools v0.0.0-20190418001031-e561f6794a2a // indirect
+	k8s.io/apimachinery v0.0.0-20190515023456-b74e4c97951f
+	k8s.io/client-go v0.0.0-00010101000000-000000000000
+)
+
+replace k8s.io/client-go => k8s.io/client-go v0.0.0-20190515063710-7b18d6600f6b
```
Comment (lieut-data) [inline go.mod:94]: Based on https://github.com/kubernetes/kubernetes/issues/76304

File: loadtest/setup_server.go [hunk 3]
```diff
@@ -197,79 +197,80 @@ func SetupServer(cfg *LoadTestConfig) (*ServerSetupData, error) {
...
-		if !*serverConfig.TeamSettings.EnableOpenServer {
-			mlog.Info("EnableOpenServer is false, attempt to set to true for the load test...")
-			*serverConfig.TeamSettings.EnableOpenServer = true
-			if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
-				mlog.Error("Failed to set EnableOpenServer", mlog.Err(resp.Error))
-				return resp.Error
-			}
+	}
+
+	if !*serverConfig.TeamSettings.EnableOpenServer {
```
Comment (lieut-data) [inline loadtest/setup_server.go:206]: Unindented here, so turn off whitespace diffs for clarity.

File: loadtest/setup_server.go [hunk 4]
```diff
@@ -197,79 +197,80 @@ func SetupServer(cfg *LoadTestConfig) (*ServerSetupData, error) {
...
+	mlog.Info(fmt.Sprintf("MaxChannelsPerTeam is %v", *serverConfig.TeamSettings.MaxChannelsPerTeam))
 
-		if !serverConfig.ServiceSettings.EnableIncomingWebhooks {
-			mlog.Info("Enabing incoming webhooks for the load test...")
-			serverConfig.ServiceSettings.EnableIncomingWebhooks = true
-			if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
-				mlog.Error("Failed to set EnableIncomingWebhooks", mlog.Err(resp.Error))
-				return resp.Error
-			}
+	if !*serverConfig.ServiceSettings.EnableIncomingWebhooks {
```
Comment (lieut-data) [inline loadtest/setup_server.go:239]: These are pointers now. The server also seems to complain about "missing" new configuration options, which was failing the loadtest @cpanato was running.

File: loadtest/tests.go [hunk 5]
```diff
@@ -42,8 +42,7 @@ type UserEntity struct {
 }
 
 func readTestFile(name string) ([]byte, error) {
-	path, _ := utils.FindDir("testfiles")
-	file, err := os.Open(path + "/" + name)
+	file, err := os.Open(name)
```
Comment (lieut-data) [inline loadtest/tests.go:45]: `utils.FindDir` no longer exists, but we didn't need all that fanciness, so I've just hard-coded the `testfiles` path instead.

**Review by crspeller — APPROVED** [review]

**Comment by lieut-data** [comment]
Note that this successfully passed a loadtest.

File: loadtest/setup_server.go (hunk 4 above)
Comment (cpanato) [reply loadtest/setup_server.go:239]: 👍

**Comment by cpanato** [comment]
@lieut-data what do you think I push the changes here https://github.com/mattermost/mattermost-load-test/pull/141 to this PR and close that one?

**Comment by lieut-data** [comment]
@cpanato, seems like we can just merge them separately? Any specific reason to want to combine the commits here?

**Comment by cpanato** [comment]
@lieut-data to make this green :)

**Comment by lieut-data** [comment]
Oh, it wasn't at all clear to me that the other PR fixes the general build issues. Let's just merge yours normally, and I'll merge in your changes before we go live with this PR.

**Review by cpanato — APPROVED** [review]

**Comment by lieut-data** [comment]
@cpanato, this is still failing the build. Going to merge and we can continue investigating that separately.

**Pull request merged**